src/credentials_rotators/npm-token-rotation/src/lambda/step-01-publish-token/index.ts

```diff
@@ -0,0 +1,96 @@
+import config from "../../../secret_config.json";
+import assert from "assert";
+import {
+  ContextVariables,
+  EnvironmentVariables,
+  updateCircleCIEnvironmentVariables,
+} from "../utils/circleci-helper";
+import * as utils from "../utils/utils";
+import {
+  NPMTokenRotationConfig,
+  TokenPublishCircleCIEnvironmentConfig,
+  TokenRotationStepFnEvent,
+} from "../../stacks/types";
+
+export const handler = async (event: TokenRotationStepFnEvent) => {
+  const tokenDetails = utils.getTokenConfigForArn(
+    config as NPMTokenRotationConfig,
+    event.secretArn
+  );
+  const webhookUrl = tokenDetails.slackWebHookConfig
+    ? await utils.getSecret(
+        tokenDetails.slackWebHookConfig.arn,
+        tokenDetails.slackWebHookConfig.secretKey
+      )
+    : undefined;
+  if (tokenDetails) {
+    try {
+      const tokenConfig = tokenDetails.publishConfig.circleCiToken;
+      if (!tokenConfig || !tokenConfig.arn || !tokenConfig.secretKey) {
+        throw Error(
+          `Invalid rotation configuration. Expected arn and token key, got ${JSON.stringify(
+            tokenConfig
+          )}`
+        );
+      }
+      const circleCIToken = await utils.getSecret(
+        tokenConfig.arn,
+        tokenConfig.secretKey
+      );
+      if (!circleCIToken) {
+        throw new Error("Could not get the CircleCI token");
+      }
+      const newNPMToken = await utils.getSecret(
+        event.secretArn,
+        tokenDetails.secretKey
+      );
+      assert(newNPMToken, "Secret manager should have newNPMToken");
+
+      let updateConfig: ContextVariables | EnvironmentVariables;
+      if (tokenDetails.publishConfig.type === "Context") {
+        updateConfig = {
+          type: tokenDetails.publishConfig.type,
+          context: tokenDetails.publishConfig.contextName!,
```

Comment:
check if `contextName` exists like the `projectName` check implemented below. Could use `TokenPublishCircleCIContextConfig` type?

Reply:
We do validation of the configuration before configuration is packaged and pushed, so this should not be a concern